utils/spdx/src/main/kotlin/parser/Exceptions.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.spdx.parser
+
+import kotlin.reflect.KClass
+
+import org.ossreviewtoolkit.utils.spdx.SpdxException
+
+/**
+ * An exception to indicate that an [SpdxExpressionLexer] error occurred.
+ */
+class SpdxExpressionLexerException(val char: Char?, val position: Int) :
```

Comment:
Why is `char` nullable? What are the semantics if it is `null`? Would "Unexpected character 'null'" then mean that the string ended earlier than expected?